src/lib/components/client/Toggle/hooks/useToggle/useToggle.ts

```diff
@@ -0,0 +1,23 @@
+"use client";
+
+import { useCallback, useState } from "react";
+
+import { UseToggleProps, UseToggleReturn } from "./useToggle.type";
+
+const useToggle = ({
+  initialState = false,
+}: UseToggleProps): UseToggleReturn => {
+  const [checked, setChecked] = useState(initialState);
+
+  const handleToggle = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setChecked(e.target.checked);
+  }, []);
+
+  return {
+    checked,
+    onChange: handleToggle,
+    setChecked,
+  };
+};
```

Comment:
```suggestion
const useToggle = ({
  initialState = false,
}: UseToggleProps)=> {
  const [checked, setChecked] = useState(initialState);

  const handleToggle = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
    setChecked(e.target.checked);
  }, []);

  return {
    checked,
    onChange: handleToggle,
    setChecked,
  } as const;
};
```

요건 팁 아닌 팁인데 위 처럼 타입을 강제로 추론 시켜줄 수 있을..걸요?

Reply:
아 ! 그렇구만요